### DISCUSSION TITLE
A plane can't refraction the mesh behind it

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#6741] [Q&A] [answered]
A plane can't refraction the mesh behind it. 
Please tell me what I am doing wrong, any advice is greatly appreciated.

![refraction](https://user-images.githubusercontent.com/3060688/232717780-e94147f1-73aa-4788-bc0a-75027a596876.JPG)

The material file：
    name : ThinRefraction,
    shadingModel : lit,
    refractionType : thin,
    refractionMode : screenspace,
    specularAntiAliasing : true,

        material.baseColor.rgb = float3(1.0,1.0,1.0);
        material.baseColor.a = 1.0;
        material.roughness = materialParams.roughness;
        material.ior = materialParams.ior;
        material.transmission = materialParams.transmission;
        material.metallic = materialParams.metallic;
        material.sheenColor = materialParams.sheenColor;
        material.sheenRoughness = materialParams.sheenRoughness;
        material.clearCoat = materialParams.clearCoat;
        material.clearCoatRoughness = materialParams.clearCoatRoughness;
        //material.anisotropy = materialParams.anisotropy;
        material.microThickness = materialParams.thickness;
        material.reflectance = 0.01;

        glassInstance.setParameter("baseColor", 1.0f, 1.0f, 1.0f)
        glassInstance.setParameter("roughness", 0.2f)
        glassInstance.setParameter("transmission", 1.0f)
        glassInstance.setParameter("ior", 1.5f)

But I find the doc can do these like below:
![refraction1](https://user-images.githubusercontent.com/3060688/232718008-5bba9e43-3a6c-49da-a71f-9ba9af3b6a8f.JPG)

**Comment by romainguy** [comment, marked as answer]
Make sure the other mesh isn't transparent.

**Reply by fanbinnet** [reply]
The other mesh inn't transparent.
You can see the bottom of the shader ball mesh.

**Reply by fanbinnet** [reply]
Thanks for the sugesstion.
I write wrong blend mode for material file.
